### ISSUE TITLE
Fixed and Variable Consumption Analysis (kWh)

### ISSUE THREAD

**Issue by LuizHenriqueVeiga** [LuizHenriqueVeiga/c-digo-portifolio-puc#1]
#Esta etapa é para importar modelos de excução
import pandas as pd
import seaborn as srn
import statistics  as sts
import matplotlib.pyplot as plt
import numpy as np
import matplotlib.pyplot as plt
import sklearn.linear_model as skl
import tabulate 
import prettytable as pt
import math as math
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.cm as cm
from datetime import datetime
from sklearn.model_selection import train_test_split, KFold, cross_validate, cross_val_score, learning_curve
from sklearn.metrics import r2_score, mean_squared_error, average_precision_score, recall_score
from statsmodels.tsa.arima_model import ARIMA
from matplotlib.patches import Rectangle
#from pmdarima.arima import auto_arima
# Display
import warnings
warnings.filterwarnings('ignore')

# Pandas options
pd.set_option('display.max_columns', 1500)
pd.set_option('display.width', 3000)
from pandas.plotting import register_matplotlib_converters
register_matplotlib_converters()

srn.set()

#Esta etapa é usada para importação de arquivo de dados 'fixoEVarivael_5.csv'
from google.colab import files
import io
df= files.upload()

#Esta etapa é usada para importação de arquivo de dados 'target_P15_8.cvs'
from google.colab import files
import io
tgP15= files.upload()

#Esta etapa é usada para importação de arquivo de dados 'geralToucador_3.csv'
from google.colab import files
import io
df1= files.upload()

#Esta etapa é usada para importação de arquivo de dados 'geralToucador_semana_1.csv'
from google.colab import files
import io
dfS= files.upload()

#Esta etapa é usada para importação de arquivo de dados 'geralToucador_fsemana_1.csv'
from google.colab import files
import io
dffS= files.upload()

#Esta etapa é usada para importação de arquivo de dados / Será Usada na Correlação
from google.colab import files
import io
dfC= files.upload()

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
df= pd.read_csv("fixoEVariavel_5.csv")

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
tgP15= pd.read_csv("target_P15_8.csv")

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
df1= pd.read_csv("geralToucador_3.csv")

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
dfS= pd.read_csv("geralToucador_semana_1.csv")

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
dffS= pd.read_csv("geralToucador_fsemana_1.csv")

#função de leitura de arquivo Dados
#checar se a extensão é csv,txt, xls ou outra
dfC= pd.read_csv("dadosCorrelacao_17.csv")

# Conversão dos atributos que estão no formato string para formato de data
dateparse = lambda dates: datetime.strptime(dates, '%d/%m/%Y %H:%M')

df

tgP15

df1

dfS

dffS

dfC

df = pd.read_csv('fixoEVariavel_5.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

tgP15 = pd.read_csv('target_P15_8.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

df1 = pd.read_csv('geralToucador_3.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

dfS = pd.read_csv('geralToucador_semana_1.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

dffS = pd.read_csv('geralToucador_fsemana_1.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

dfC = pd.read_csv('dadosCorrelacao_17.csv', parse_dates = ['Time'],index_col = 'Time', date_parser = dateparse)

#função para ver o tipo do arquivo - tem que ser float, int....
df.dtypes

#função para ver o tipo do arquivo - tem que ser float, int....
tgP15.dtypes

#função para ver o tipo do arquivo - tem que ser float, int....
df1.dtypes

#função para ver o tipo do arquivo - tem que ser float, int....
dfS.dtypes

#função para ver o tipo do arquivo - tem que ser float, int....
dffS.dtypes

#função para ver o tipo do arquivo - tem que ser float, int....
dfC.dtypes

df.index

tgP15.index

df1.index

dfS.index

dffS.index

dfC.index

tgP15

dfS

dffS

dfC

#Função para descrever o período de verao_2021_1 = Janeiro - Março
verao_2021 = df['2021-01-01 00:00:00':'2021-03-21 00:00:00']

#Função para descrever o período de verao_2021_2 = Dezembro - Março
verao_2021_2022 = df['2021-12-21 00:00:00':'2022-03-21 00:00:00']

#Função para descrever o período de outono_2021 = Março - Junho
outono_2021 = df['2021-03-21 00:00:00':'2021-06-21 00:00:00']

#Função para descrever o período de outono_2022 = Março - Junho
outono_2022 = df['2022-03-21 00:00:00':'2022-06-21 00:00:00']

#Função para descrever o período de inverno_2021 = Junho - Setembro
inverno_2021 = df['2021-06-21 00:00:00':'2021-09-23 00:00:00'] 

#Função para descrever o período de inverno_2021 = Junho - Setembro
inverno_2022 = df['2022-06-21 00:00:00':'2022-09-23 00:00:00']   

#Função para descrever o período de primavera_2021 = Setembro - Dezembro
primavera_2021 = df['2021-09-23 00:00:00':'2021-12-21 00:00:00']    

df.DJ_01_P15.describe(), df.DJ_02_P15.describe(), df.DJ_03_P15.describe(), df.DJ_04_P15.describe(), df.DJ_05_P15.describe(), df.DJ_06_P15.describe(), df.DJ_07_P15.describe(), df.DJ_08_P15.describe()

tgP15.TG_01_P15.describe(), tgP15.TG_02_P15.describe(), tgP15.TG_03_P15.describe(), tgP15.TG_04_P15.describe(), tgP15.TG_05_P15.describe(), tgP15.TG_06_P15.describe(), tgP15.TG_07_P15.describe(), tgP15.TG_08_P15.describe()

dfS.FX_15_kWh.describe(), dfS.FX_15_m3h.describe(), dfS.FX_15_m3.describe()

dffS.FX_15_kWh.describe(), dffS.FX_15_m3h.describe(), dffS.FX_15_m3.describe()

dfC.DJ_01.describe(), dfC.DJ_02.describe(), dfC.DJ_03.describe(), dfC.DJ_05.describe(), dfC.DJ_06.describe(), dfC.DJ_08.describe(), dfC.GZ_8A.describe(), dfC.GZ_9A.describe(), dfC.GZ_9B.describe(), dfC.C2.describe(), dfC.C3.describe(), dfC.LS1.describe(), dfC.LS2.describe(), dfC.LS3.describe(), dfC.LS7.describe(), dfC.LS8.describe(), dfC.LS9.describe(),  dfC.T01.describe(), dfC.STM.describe(), dfC.CDA.describe()

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_01_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,1)
plt.plot(df.DJ_01_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_01_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_01_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_02_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,2)
plt.plot(df.DJ_02_P15,color = "violet", linestyle ="solid" )
plt.plot(tgP15.TG_02_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_02_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_03_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,3)
plt.plot(df.DJ_03_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_03_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_03_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_04_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,4)
plt.plot(df.DJ_04_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_04_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_04_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_05_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,5)
plt.plot(df.DJ_05_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_05_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_05_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_06_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,6)
plt.plot(df.DJ_06_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_06_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_06_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_07_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,7)
plt.plot(df.DJ_07_P15,color = "violet", linestyle = "solid")
plt.plot(tgP15.TG_07_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_07_P15')

plt.figure()
plt.figure(figsize=(30,20))

#Consumo (kW) DJ_08_P15 = Medidor EE_SE_08_SI.BARRAMENTO_440V_TOUCADOR Real Power (kW)
plt.subplot(8,1,8)
plt.plot(df.DJ_08_P15,color = "violet", linestyle = "solid" )
plt.plot(tgP15.TG_08_P15,color = "blue", linestyle = "solid" )
plt.title('DJ_08_P15')

plt.figure()
plt.figure(figsize=(30,30))

#Consumo (kWh) FX_15_Toucador = Medidor Geral de Consumo de Toucador - Ilustrando Semana e Finais de Semana
plt.subplot(2,1,1)
plt.plot(dfS.FX_15_kWh,color = "darkblue",marker="s",markersize=5, linestyle = "dotted", label = "Semana" )
plt.plot(dffS.FX_15_kWh,color = "blue",marker="s",markersize=5, linestyle = "dotted", label = "Final de Semana")
plt.title('FX_15_Toucador')
plt.legend()

plt.figure()
plt.figure(figsize=(30,3))

plt.subplot(1,4,1)
srn.distplot(verao_2021.iloc[:,1], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Verão 2021 - Jan-Mar')

plt.subplot(1,4,2)
srn.distplot(verao_2021.iloc[:,2], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Verão 2021 - Jan-Mar')

plt.subplot(1,4,3)
srn.distplot(verao_2021.iloc[:,3], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Verão 2021 - Jan-Mar')

plt.subplot(1,4,4)
srn.distplot(verao_2021.iloc[:,6], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Verão 2021 - Jan-Mar')

plt.figure()
plt.figure(figsize=(30,3))

plt.subplot(1,4,1)
srn.distplot(outono_2021.iloc[:,1], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Outono 2021 - Mar-Jun')

plt.subplot(1,4,2)
srn.distplot(outono_2021.iloc[:,2], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Outono 2021 - Mar-Jun')

plt.subplot(1,4,3)
srn.distplot(outono_2021.iloc[:,3], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Outono 2021 - Mar-Jun')

plt.subplot(1,4,4)
srn.distplot(outono_2021.iloc[:,6], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Outono 2021 - Mar-Jun')

plt.figure()
plt.figure(figsize=(30,3))

plt.subplot(1,4,1)
srn.distplot(inverno_2021.iloc[:,1], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Inverno 2021 - Jun-Set')

plt.subplot(1,4,2)
srn.distplot(inverno_2021.iloc[:,2], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Inverno 2021 - Jun-Set')

plt.subplot(1,4,3)
srn.distplot(inverno_2021.iloc[:,3], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Inverno 2021 - Jun-Set')

plt.subplot(1,4,4)
srn.distplot(inverno_2021.iloc[:,6], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Inverno 2021 - Jun-Set')

plt.figure()
plt.figure(figsize=(30,3))

plt.subplot(1,4,1)
srn.distplot(primavera_2021.iloc[:,1], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Primavera 2021 - Set-Dez')

plt.subplot(1,4,2)
srn.distplot(primavera_2021.iloc[:,2], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Primavera 2021 - Set-Dez')

plt.subplot(1,4,3)
srn.distplot(primavera_2021.iloc[:,3], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Primavera 2021 - Set-Dez')

plt.subplot(1,4,4)
srn.distplot(primavera_2021.iloc[:,6], hist = True, kde = True, bins = 80, color = 'blue',hist_kws={'edgecolor': 'black'})
plt.title('Primavera 2021 - Set-Dez')

df.shape

plt.figure(figsize=(18, 10))
srn.heatmap(dfC.corr(),  cmap="coolwarm", vmin = -1, vmax = 1,
annot = True, fmt='.2f', annot_kws = {'size' : 10, 'color' : 'black'})

plt.figure(figsize=(10,8))
srn.regplot(dfC.DJ_G, dfC.STM, x_jitter = 0.0, fit_reg = False)
